### ISSUE TITLE
Monitor.signal_done called after instantiation is awkward

### ISSUE THREAD

**Issue by jondequinor** [equinor/ert#1227]
There's a difference between
```python
mon = _Monitor(host, port)
for _ in mon.track():
    pass
mon.signal_done()
```
and
```python
_Monitor(host, port).signal_done()
```

I think both are legitimate, but `track` starts the loop, whereas `exit_server` requires a running loop. Maybe disregard the loop and make `exit_server` async?

**Comment by DanSava** [comment]
Is this issue still relevant. 

There is no instance of `Monitor.exit_server()` in the current version of the code. 

Or am I misunderstanding this issue and the point is that we need to add `exit_server()` to the Monitor?

**Comment by jondequinor** [comment]
It's renamed to `signal_done`

**Comment by DanSava** [comment]
still unclear what needs to be done for this issue to be solved. 

Do you just want to make `signal_done` to be an async method? If so is that really a good idea? Not really clear what it solves.

**Comment by jondequinor** [comment]
Yes, it is unclear what needs to be done, hence the issue… 

What you propose at least kills this code ```python
        if self._loop.is_running():
            asyncio.run_coroutine_threadsafe(_send(), self._loop).result()
        else:
            self._loop.run_until_complete(_send())```

I would be happy if that was the case. That code is bad.

**Comment by markusdregi** [comment]
Related to #1332

**Comment by verveerpj** [comment]
This may be (partially) resolved by pull request #1428. I will look into the usage of `signal_done` throughout the code and resolve this issue.

**Comment by verveerpj** [comment]
Seems to be resolved by #1428
